src/foundry/client-esm/dice/roll.d.mts

```diff
@@ -0,0 +1,636 @@
+
+  /**
+   * Recreate a Roll instance using a provided JSON string
+   * @param json - Serialized JSON data representing the Roll
+   * @returns A reconstructed Roll instance
+   */
+  static fromJSON(json: string): Roll;
+
+  /**
+   * Manually construct a Roll object by providing an explicit set of input terms
+   * @param terms -The array of terms to use as the basis for the Roll
+   * @param options - Additional options passed to the Roll constructor
+   * @returns The constructed Roll instance
+   *
+   * @example Construct a Roll instance from an array of component terms
+   * ```typescript
+   * const t1 = new Die({number: 4, faces: 8};
+   * const plus = new OperatorTerm({operator: "+"});
+   * const t2 = new NumericTerm({number: 8});
+   * const roll = Roll.fromTerms([t1, plus, t2]);
+   * roll.formula; // 4d8 + 8
+   * ```
+   */
+  static fromTerms<T extends ConstructorOf<Roll<any>>>(
+    this: T,
+    terms: RollTerm[],
+    options?: InexactPartial<Roll.Options>,
+  ): InstanceType<T>;
+}
+
+declare namespace Roll {
+  interface Options extends RollTerm.EvaluationOptions {
+    /**
+     * If false, force the use of non-interactive rolls and do not prompt the user to make manual rolls.
+     * @defaultValue `true`
+     */
+    allowInteractive?: boolean;
+    /**
+     * Throw an Error if the Roll contains non-deterministic terms that cannot be evaluated synchronously.
+     *  If this is set to false, non-deterministic terms will be ignored.
+     */
+    strict?: boolean;
+  }
+
+  interface SplitGroupOptions {
+    openRegexp: RegExp | string;
+    closeRegexp: RegExp | string;
+    openSymbol: string;
+    closeSymbol: string;
+    onClose: (group: { open: string; terms: string[]; close: string }) => string[];
+  }
+
+  /** Additional options to configure how the link is constructed. */
+  interface ToAnchorOptions {
+    /** A custom label for the total. */
+    label: string;
+
+    /** Attributes to set on the link. (default: `{}`) */
+    attrs: Record<string, string>;
+
+    /** Custom data attributes to set on the link. (default: `{}`) */
+    dataset: Record<string, string | undefined>;
+
+    /**
+     * Additional Classes to add to the link. (default: `[]`)
+     * The classes `inline-roll` and `inline-result` are added by default.
+     */
+    classes: string[];
+
+    /** A font-awesome icon class to use as the icon instead of a d20. */
+    icon: string;
+  }
+
+  interface Data {
+    formula: string;
+    results: Array<number | string>;
+    terms: Array<PoolTerm.TermData | DiceTerm.Data>;
+    total: number | null;
+  }
+
+  type Flavor = Record<`%F${number}%`, string>;
+
+  type MessageData<T extends DeepPartial<ConstructorParameters<typeof ChatMessage>[0]>> = {
+    user: string;
+    type: (typeof foundry.CONST.CHAT_MESSAGE_TYPES)["ROLL"]; // TODO: Update this
+    content: number;
+    sound: typeof CONFIG.sounds.dice;
+  } & T;
```

Comment:
Nit; you probably want it to be `T & { ... }`. More for consistency but also partially for evaluation ordering.